### PR TITLE
Add detailed class diagram for commands and parsers in DG

### PR DESCRIPTION
**Pull request by WuHaohui1231**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -113,7 +113,22 @@ How the parsing works:
...
 * All `XYZCommandParser` classes (e.g., `AddCommandParser`, `DeleteCommandParser`, ...) inherit from the `Parser` interface so that they can be treated similarly where possible e.g, during testing.
 
+
+### Command Classes
+
+The class diagram below expands the details of gCommand and Parser part in the Logic component above, showing the details of how commands are parsed and created
+
+Simple commands without arguments including `clear` `list` `exit` `help` are created directly by `AddressBookParser`<br/>
+To parse complex commands with arguments, including `add` `find` `edit` `delete`, `AddressBookParser` will create customized parser corresponding to the command. <br/>
+Then the customized parser parse the arguments and create the command
```
Comment (bensohh) [inline docs/DeveloperGuide.md]: I think it should be the customized parser parses the arguments...

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -113,7 +113,22 @@ How the parsing works:
...
+
+The diagram also included some new classes involved, like the `find` commands depends on new predicates in the `Model` component to allow all-info and fuzzy search (more detail in the `find` command description)
+
```
Comment (Samsation) [inline docs/DeveloperGuide.md:126]: I like how you differentiated the various types of commands and how they are being processed by the Parsers. 😸

**Review by bensohh — APPROVED** [review]
LGTM! Just some minor grammar/spelling errors! 👍

File: docs/diagrams/CommandClasses.puml [hunk 3]
```diff
@@ -0,0 +1,107 @@
+@startuml
+'!include style.puml
+'skinparam arrowThickness 1.1
+'skinparam arrowColor UI_COLOR_T4
+'skinparam classBackgroundColor UI_COLOR
```
Comment (Samsation) [inline docs/diagrams/CommandClasses.puml:5]: Perhaps you could change the color to match the boba theme (similar to the activity diagrams) 🧋

**Review by Samsation — APPROVED** [review]
LGTM! 🔥

**Pull request merged**